cython/README.md

```diff
@@ -1,33 +1,36 @@
     - If the interpreter is inside an environment (such as an anaconda environment or virtualenv environment) then the environment should be active while building gtsam.
-- This wrapper needs Cython(>=0.25.2), backports_abc>=0.5, and numpy. These can be installed as follows:
+- This wrapper needs `Cython(>=0.25.2)`, `backports_abc(>=0.5)`, and `numpy(>=1.11.0)`. These can be installed as follows:
 
-```bash
- pip install -r <gtsam_folder>/cython/requirements.txt
-```
+  ```bash
+  pip install -r <gtsam_folder>/cython/requirements.txt
+  ```
 
 - For compatibility with gtsam's Eigen version, it contains its own cloned version of [Eigency](https://github.com/wouterboomsma/eigency.git),
-named **gtsam_eigency**, to interface between C++'s Eigen and Python's numpy.
+named `gtsam_eigency`, to interface between C++'s Eigen and Python's numpy.
+
+## Install
 
-- Build and install gtsam using cmake with `GTSAM_INSTALL_CYTHON_TOOLBOX` enabled.
-The wrapped module will be installed to `GTSAM_CYTHON_INSTALL_PATH`, which is
-by default: `<your CMAKE_INSTALL_PREFIX>/cython`
+- Run cmake with the `GTSAM_INSTALL_CYTHON_TOOLBOX` cmake flag enabled to configure building the wrapper. The wrapped module will be built and copied to the directory defined by `GTSAM_CYTHON_INSTALL_PATH`, which is by default: `<CMAKE_INSTALL_PREFIX>/cython.build`.
 
 - To use the library without installing system-wide: modify your `PYTHONPATH` to include the `GTSAM_CYTHON_INSTALL_PATH`:
-```bash
-export PYTHONPATH=$PYTHONPATH:<GTSAM_CYTHON_INSTALL_PATH>
-```
-- To install system-wide: run `make install` then navigate to `GTSAM_CYTHON_INSTALL_PATH` and run `python setup.py install`
-    - (the same command can be used to install into a virtual environment if it is active)
-    - note: if you don't want gtsam to install to a system directory such as `/usr/local`, pass `-DCMAKE_INSTALL_PREFIX="./install"` to cmake to install gtsam to a subdirectory of the build directory.
-    - if you run `setup.py` from the build directory rather than the installation directory, the script will warn you with the message: `setup.py is being run from an unexpected location`.
-      Before `make install` is run, not all the components of the package have been copied across, so running `setup.py` from the build directory would result in an incomplete package.
+
+  ```bash
+  export PYTHONPATH=$PYTHONPATH:<GTSAM_CYTHON_INSTALL_PATH>
+  ```
+
+- Build GTSAM and the wrapper with `make`.
+
+- To install system-wide, simply run `make install`.
+    - The same command can be used to install into a virtual environment if it is active.
+    - **NOTE**: if you don't want gtsam to install to a system directory such as `/usr/local`, pass `-DCMAKE_INSTALL_PREFIX="./install"` to cmake to install gtsam to a subdirectory of the build directory.
```

Comment:
Awesome, but still the concern: this PR will always **globally** install the Cython wrapper, regardless of the prefix specified, when running `make install`?

Reply:
Actually that's a mistake in the docs. The concept of the install prefix for the python wrapper is no longer valid since it builds everything in the CMake build directory. This makes everything self contained, and the install command installs the python package to the python namespace.

Reply:
I still insist that `make install` should not touch anything beyond the prefix, including the current `site-packages` directory. This is a common practice and we should not break it.

Reply:
First of all, I always use a prefix, and it works like a charm, so I understand the documentation to be correct. And indeed, the fact that cython is installed outside the prefix as Fan mentions, does seem like a breach of contract. We should probably change the default to something inside the prefix.

Reply:
By the way, Varun, your use of prefix seems unconventional. I always use my home directory as the prefix, so I have an include and lib folder there.

Reply:
I was unclear about what I meant in my previous comment. That was *my* mistake in the docs. The way this PR does this is that cython is no longer installed to the prefix location, so the notion of a prefix is no longer applicable. Everything happens inside the CMake build directory.

And when I mentioned the package, I mean the result of running python setup.py install which is technically what installs the GTSAM package and makes it available via python.

Thanks to this PR, the only things that depend on the prefix are the C++ libraries and the Matlab wrapper.

Reply:
I do think in line with Varun that the cython wrapper should not be installed to the prefix in its current subdirectory - it should be inside `$PREFIX/share/gtsam/cython` or something, or **NOT** installed at all (only reside in build directory), but I do **NOT** think that `make install` should install anything to the **system** (`site-packages`, etc.), either.

Reply:
I actually prefer the **NOT** installed at all option. I am still being mindful of build postfixes so that shouldn't be a problem.